src/artkit/model/llm/vertexai/_vertexai.py

```diff
@@ -85,14 +86,32 @@ class VertexAIChat(ChatModelConnector[GenerativeModel], metaclass=ABCMeta):
     """
 
     region: str | None
-    gcp_project_id: str | None
+    gcp_project_id_env: str | None
     #: If ``False``, disable all accessible safety categories; otherwise, enable them.
     safety: bool
 
     @classmethod
     def get_default_api_key_env(cls) -> str:
         """[see superclass]"""
         return ""
+    
+    def get_gcp_project_id_env(self) -> str:
+        """
+        Get the GCP project ID from the environment variable specified by 
+        :attr:`gcp_project_id_env`.
+
+        :return: the GCP project ID
+        :raises ValueError: if the environment variable is not set
+        """
+        try:
+            return os.environ[self.gcp_project_id_env]
+        except KeyError as e:
+            raise ValueError(
```

Comment:
Suggest we raise an `EnvironmentError` instead.

Reply:
I copied this pattern from the `get_api_key` method of the `ConnectorMixin` class: https://github.com/BCG-X-Official/artkit/blob/1.0.x/src/artkit/model/base/_model.py#L231

Shall I change it there too?